File: ImgProc/tutorial_pyramids/Pyramids.cpp
/**
 * @file Pyramids.cpp
 * @brief Sample code of image pyramids (pyrDown and pyrUp)
 * @author OpenCV team
 */

#include "opencv2/imgproc/imgproc.hpp"
#include "opencv2/imgcodecs.hpp"
#include "opencv2/highgui/highgui.hpp"
#include <math.h>
#include <stdlib.h>
#include <stdio.h>

using namespace cv;

Mat img;
const char* window_name = "Pyramids Demo";

int main(void)
{
    printf("\n Zoom In-Out demo  \n ");
    printf("------------------ \n");
    printf(" * [u] -> Zoom in  \n");
    printf(" * [d] -> Zoom out \n");
    printf(" * [ESC] -> Close program \n \n");

    //! [load_image]
    img = imread("../data/chicky_512.png");
    if (img.empty()) {
        printf(" No data! -- Exiting the program \n");
        return -1;
    }
    //! [load_image]

    //! [create_window]
    namedWindow(window_name, WINDOW_AUTOSIZE);
    imshow(window_name, img);
    //! [create_window]

    //! [loop]
    for (;;) {
        int c;
        c = waitKey(10);
        if ((char)c == 27) {
            break;
        }
        if ((char)c == 'u') {
            //! [pyrUp]
            pyrUp(img, img, Size(img.cols * 2, img.rows * 2));
            //! [pyrUp]
            printf("** Zoom In: Image x 2 \n");
        }
        else if ((char)c == 'd') {
            //! [pyrDown]
            pyrDown(img, img, Size(img.cols / 2, img.rows / 2));
            //! [pyrDown]
            printf("** Zoom Out: Image / 2 \n");
        }
        imshow(window_name, img);
    }
    //! [loop]
    return 0;
}

File: ImgProc/tutorial_pyramids/Pyramids.py
import cv2

print " \nZoom In-Out demo "
print " ------------------ "
print "  * [u] -> Zoom in  "
print "  * [d] -> Zoom out "
print "  * [ESC] -> Close program \n"

img = cv2.imread('../data/chicky_512.png')

while 1:
    h, w = img.shape[:2]

    cv2.imshow('Pyramids Demo', img)
    k = cv2.waitKey(10) % 256

    if k == 27:
        break

    elif k == ord('u'):  # Zoom in, make image double size
        img = cv2.pyrUp(img, dstsize=(2 * w, 2 * h))

    elif k == ord('d'):  # Zoom down, make image half the size
        img = cv2.pyrDown(img, dstsize=(w / 2, h / 2))

cv2.destroyAllWindows()
